### ISSUE TITLE
Confusing results after step-2 training.

### ISSUE THREAD

**Issue by wutong16** [VITA-Group/SinNeRF#10]
Hi! Thanks for your great work and implementation!

I'm currently trying your code on nerf_synthetic (lego) and dtu (scan4), while the results after the two-stage training are confusing. 

Specifically, the evaluation psnr for lego is 20.6 by the end of step 1, while it drops to 14.9 after step 2 training. The same thing happens to dtu_scan_4, where the evaluation psnr is around 15.0 after step 1 and drops to 11.8 after step 2. The visualization results for lego are as below. 

I'd appreciate it if you could provide some idea on this phenomenon and how to fix this problem. Thank you!

lego after step 1
![002](https://user-images.githubusercontent.com/108319439/176104949-0e4d096f-9fc1-47d5-8dfc-9691e3d63bf8.png)
lego after step 2
![010](https://user-images.githubusercontent.com/108319439/176104904-799d592e-4dc0-44c8-aed9-9a511a74d048.png)

**Comment by Ir1d** [comment]
@wutong16 Hi, could you please try running step2 with `--depth_smooth_weight 0` ?

**Comment by wutong16** [comment]
Many thanks for your prompt reply! 

Sorry that I found a mistake in my previous dtu setting, which led to the 11.9 performance for step 2; after fixing it, it becomes around 13.5. 

I then tried `--depth_smooth_weight 0` for them, while the psnr results are quite close to the original setting. 
- For dtu_scan4, I think that the appearance is indeed visually better after training with the discriminator in step 2, but the psnr becomes lower. 
- For lego, there is a strange hole after step 2, whether we use `--depth_smooth_weight 0` or not. 

I'm not sure what caused the problem above... is the code running well on your device?

**Comment by Ir1d** [comment]
Thanks for letting me know, im investigating the issue now. There might be tiny bugs when I try to tidy the code from my previous codebase

**Comment by Ir1d** [comment]
Hi @wutong16, I've pushed some minor update to the code. Could you please try again by adding `--angle=20`? Training stage2 on a smaller angle (compared to 30) will help stabilize the GAN training. Also, using a smaller learning rate (e.g. 5e-5) and smaller loss weight for discriminator (e.g. 0.01) helps.

Here's some preliminary result (at 300 epoch of stage2) with PSNR 21.03 (better than 20.97 reported in paper):

![test](https://user-images.githubusercontent.com/10709657/177248249-9f1f96da-b72f-4710-b191-deebd7453474.gif)



btw, the previous code also might have some nan issue (in warping function in dataloader) under certain numpy versions, did you use packages similar to the environment.yaml in this repo?

**Comment by wutong16** [comment]
Hi @Ir1d, many thanks for the update! I'll try the new code and config. 

Yes, I've been using the environment created with the `environment.yaml` in this repo.

**Comment by happysxpp** [comment]
> btw, the previous code also might have some nan issue (in warping function in dataloader) under certain numpy versions, did you use packages similar to the environment.yaml in this repo?

@Ir1d Hi, thanks a lot for sharing the code and your great work!

But when I trained on `nerf_synthetic(lego, hotdog)` , I encountered the problem of `loss=nan`.

The environment I created with the `environment.yaml`  you provided. The numpy version is the same as that in the yaml file.
```
SinNeRF$ conda list | grep numpy

numpy                     1.21.2           py37h20f2e39_0    defaults
numpy-base                1.21.2           py37h79a1101_0    defaults
```

If you can provide some suggestions to deal with this problem, I will be very grateful！

**Comment by Ir1d** [comment]
@happysxpp Thanks for your experiments. Could you please try again with the latest code? Note that NeRF training and GAN training are sensitive to initialization. Restarting the training might be a good idea if you find the loss too large at the beginning of training. If it's still not successful, please open a new issue and share your logs. Thanks!
